smf/clickhouse_keeper/method_script.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -x
+set -o errexit
+set -o pipefail
+
+. /lib/svc/share/smf_include.sh
+
+LISTEN_ADDR="$(svcprop -c -p config/listen_addr "${SMF_FMRI}")"
+LISTEN_PORT="$(svcprop -c -p config/listen_port "${SMF_FMRI}")"
+DATASTORE="$(svcprop -c -p config/store "${SMF_FMRI}")"
+DATALINK="$(svcprop -c -p config/datalink "${SMF_FMRI}")"
+GATEWAY="$(svcprop -c -p config/gateway "${SMF_FMRI}")"
+
+if [[ $DATALINK == unknown ]] || [[ $GATEWAY == unknown ]]; then
+    printf 'ERROR: missing datalink or gateway\n' >&2
+    exit "$SMF_EXIT_ERR_CONFIG"
+fi
+
+# TODO remove when https://github.com/oxidecomputer/stlouis/issues/435 is addressed
+ipadm delete-if "$DATALINK" || true
+ipadm create-if -t "$DATALINK"
+
+ipadm set-ifprop -t -p mtu=9000 -m ipv4 "$DATALINK"
+ipadm set-ifprop -t -p mtu=9000 -m ipv6 "$DATALINK"
+
+ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll"
+ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
+route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
+
+# Retrieve addresses of the other clickhouse nodes, order them and assign them to be a replica or keeper node.
+# In a follow up PR, keepers will be their own service.
+#
+# This should probably be 5 keepers?
+CH_ADDRS="$(/opt/oxide/internal-dns-cli/bin/dnswait clickhouse-keeper -H \
```

Comment:
I'm not sure we need to fix this now, but worth considering:

Some day, in the near future, we'll be trying to cope with a system where sleds (and disks) are added and removed from the rack. In this scenario, the following could happen:

- We start with this set of 3 keepers
- One of the three keepers dies
- Nexus decides to provision a new keeper somewhere else

For the "original" two keepers, how do they coordinate with the newly provisioned third keeper? Would this require restarting **all** keeper services with the new IPs?

Reply:
Yes, we will sadly need to restart all keepers. I'm assuming they would have a new hostname as well, which means we'd have to restart the replicas as well :/